install/0000_30_machine-api-operator_00_credentials-request.yaml

```
@@ -0,0 +1,31 @@
+apiVersion: cloudcredential.openshift.io/v1beta1
+kind: CredentialsRequest
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: openshift-cluster-api
+  namespace: openshift-cloud-credential-operator
+spec:
+  secretRef:
+    name: aws-cloud-credentials
+    namespace: openshift-cluster-api
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1beta1
+    kind: AWSProviderSpec
+    statementEntries:
+    - effect: Allow
+      action:
+      - ec2:DescribeImages
+      - ec2:DescribeVpcs
+      - ec2:DescribeSubnets
+      - ec2:DescribeAvailabilityZones
+      - ec2:DescribeSecurityGroups
```

We're missing permission for `ec2:DescribeVolumes` [here](https://github.com/openshift/cluster-api-provider-aws/blob/ee2a7ec65a12e286655e9dc06c3e7d7f7151681b/pkg/client/client.go#L106).

```suggestion
      - ec2:DescribeSecurityGroups
      - ec2:DescribeVolumes
```

---

can you elaborate why we need it? thanks!

---

Spoke with @bison, when we switch to the credential request this would break the e2e tests because we're invoking DescribeVolumes from there if I'm not mistaken.

---

Dropping the change request because it's only needed in the e2e tests. We've to figure out a way how to get those permissions when running them.

---

yes, these are the perms to be used by the actuator. Not sure which e2e test you are refering but any of https://github.com/openshift/cluster-api-provider-aws/tree/master/test use a CI account.
Also we shouldn't grant perms inside the cluster for the sake of being able to run our test